### ISSUE TITLE
[OSD] On-screen amperage meter readings don't seem to be correct

### ISSUE THREAD

**Issue by SandroGrzicic** [cleanflight/cleanflight#2480]
On the SP Racing F3 OSD/PDB, after wiring up everything and testing that the OSD works, everything seems correct*, except the on-screen current meter, which doesn't seem to work properly. With the main voltage supply connected, **it is showing 0.07A, sometimes going up to 0.10A**. With only the USB supply connected, it shows 0.05A. These values cannot be correct since the PDB/OSD board itself, VTX, camera, FC (SP Racing F3 Evo with latest Betaflight) and an RX (FrSky XSR) surely consume more than 100mA. **After arming the quad and starting motors, the current readings still don't change.**

I have tested this with two separate boards and the same behaviour persists. Both are freshly flashed with the latest Cleanflight v1.14.0. Please close this if this is a hardware issue, I am opening it in case it's a software issue.

(*I did a quick check of modes and I think I saw that when I switched to HORIZON mode, I saw "ALT" in the OSD, and when I also enabled BARO I saw "ALT MAG", while in MAG nothing was shown, I suppose that's just a quirk of Betaflight sending unexpected data, in any case it's a super minor issue and these elements can be removed from the OSD overlay anyway.)

**Comment by element51** [comment]
i have the exact amperage issue with my SPRacingF3OSD, voltage is reading ok.

**Comment by SandroGrzicic** [comment]
Any updates on this? I would like to try fixing the issue, but without knowing where approximately in the codebase to look at, it's a bit difficult (I have already checked the amperage sensor code, I don't see anything obviously wrong there).

Some random ideas (assuming the issue is in Cleanflight and not in the board itself):

- maybe the "raw" ADC readings come from the virtual current sensor? But that doesn't explain the random jitter, nor the difference between USB plugged/unplugged.
- maybe the current sensor's current readings are corrupted by Betaflight FC data? But that doesn't explain seeing the exact same behaviour when the FC is disconnected (surely the PDB board doesn't need the FC to display the amperage).
- `adcGetChannel` might be using the wrong channel (`ADC_AMPERAGE`)? I noticed the same method is used to get the voltage readings; maybe instead of the current, the code is actually reading one of the voltage ADCs (e.g. the 5V or 12V voltages, the main battery voltage, or the 3.3V reg voltage)?
- the wrong ADC driver is being used?
